the-collab-lab/tcl-17-smart-shopping-list
# Add comment with issue AC to project.

## Description
This PR adds the `PropTypes` dependency and allows the user to click on a button and will utilize the script in `src/lib/tokens.js` to generate a token and save it to `Local Storage`



## Related Issue
[#4](https://github.com/the-collab-lab/tcl-17-smart-shopping-list/issues/4)



## Acceptance Criteria
- [x] Clicking the button/link generates a new token and saves it to `Local Storage`
- [x] For users who do have a token already saved in `Local Storage`, they should be shown/redirected to the "list" view



## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|  ✓ | :sparkles: New feature     |


## Updates
From the home screen we created a `Create a New List` button to generate a token and save the token to `Local Storage`. Once created the user is redirected to the list view. If the token already exists in `Local Storage` the user is redirected to the list view.

### Before

<img width="382" alt="Screen Shot 2021-01-06 at 11 49 41 AM" src="https://user-images.githubusercontent.com/14856004/104389650-90f16700-5501-11eb-87e5-6f6453a9c3df.png">

### After

![Screen Shot 2021-01-12 at 6 10 00 PM](https://user-images.githubusercontent.com/14856004/104389618-7e772d80-5501-11eb-9d1e-c64b210fb561.png)

## Testing Steps / QA Criteria


* One way to test the site is to click on the `details` button below to open the Netlify preview

![Screen Shot 2021-01-12 at 6 05 51 PM](https://user-images.githubusercontent.com/14856004/104389400-ff81f500-5500-11eb-95ae-4c0169a069cb.png)

* Otherwise from your terminal, pull down this branch with `git pull origin BL-VT-generate-token-new-list` and check that branch out with `git checkout BL-VT-generate-token-new-list`
* install new dependencies with `npm install`
* `npm start` to launch the app.
* Click the `Create a New List` button to generate a token and go to List view
* To see the token in `Local Storage`, open developer tools and click on `Application` tab. Find the token under the `Local Storage` dropdown menu on left side of the screen

![Screen Shot 2021-01-12 at 6 04 41 PM](https://user-images.githubusercontent.com/14856004/104389282-b3cf4b80-5500-11eb-9ce6-3c4741c87bd9.png)
